app/models/metric/capture.rb

```diff
@@ -168,37 +170,19 @@ def self.calc_tasks_by_rollup_parent(targets_by_rollup_parent)
           :start    => task_start_time,
           :end      => task_end_time,
           :parent   => pkey,
-          :targets  => targets_by_rollup_parent[pkey],
+          :targets  => targets_by_rollup_parent[pkey].map { |target| "#{target.class}:#{target.id}" },
           :complete => [],
           :interval => "realtime"
         }
       )
       _log.info "Created task id: [#{task.id}] for: [#{pkey}] with targets: #{targets_by_rollup_parent[pkey].inspect} for time range: [#{task_start_time} - #{task_end_time}]"
-      h[pkey] = task
-      h
-    end
-
-    tasks_by_rollup_parent
-  end
-  private_class_method :calc_tasks_by_rollup_parent
-
-  def self.calc_target_options(zone, targets, targets_by_rollup_parent, tasks_by_rollup_parent)
-    targets.each_with_object({}) do |target, all_options|
-      interval_name = perf_target_to_interval_name(target)
-
-      options = {:zone => zone}
-      target.perf_rollup_parents(interval_name).to_a.compact.each do |parent|
-        if tasks_by_rollup_parent.key?("#{parent.class}:#{parent.id}")
-          pkey = "#{parent.class}:#{parent.id}"
-          tkey = "#{target.class}:#{target.id}"
-          if targets_by_rollup_parent[pkey].include?(tkey)
-            # FIXME: check that this is still correct
```

Comment:
@kbrock yes I remember that for this part I couldn't do a full test because I didn't know what to expect there in all cases.
I don't remember if there were enough spec tests.